packages/contracts/contracts/registry/ERC165ContractRegistry.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.10;
+
+import "../core/component/Component.sol";
+
+/// @title An ERC165-based registry for contracts
+/// @author Aragon Association - 2022
+/// @notice This contract allows to register contracts
+contract ERC165ContractRegistry is Component {
```

Comment:
There was a very similar class on @Rekard0's PR. Aren't we solving the same problem twice? :-/ 

[packages/contracts/contracts/core/component/ERC165Registry.sol](https://github.com/aragon/core/pull/53/files#diff-cea6f9aadf6b753351fff1c1580aff9b01d050cf6a1dcc2d8a65657e14d01263)

Reply:
This feels more compact and simple, but by inheriting from `Component` we're opting into some features by default that I'm not sure we need right now (upgradeability, etc)

Reply:
> Aren't we solving the same problem twice
Yes, Sarkawt copied and adapted my code on his branch. This is what I talked about in the daily meeting today and why I suggested waiting with the review.

Reply:
> Note that in @Rekard0's version, ERC165Registry.sol is intentionally upgradable and also has permissions (as Components do).

Agree, hence my suggestion to only add the minimal required extra features :)

> Moreover, this is a contract that developers want to reuse (we can use it 2-3 times in our own DAO creation and registration process) and therefore it should be a Component IMO.

But again, I'm curious to understand the use case for this contract